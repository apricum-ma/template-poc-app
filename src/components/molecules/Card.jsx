import React, { PureComponent } from 'react'
import styled from 'styled-components'

const CardContainer = styled.div`
  position: relative;
  min-height: 450px;
  width: 300px;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border: 1px solid #ededed;
  background-color: #fff;
  box-shadow: 1px 2px 10px rgba(0,0,0,.1);
  transition: all .2s ease-in-out;
  cursor: pointer;
  &:hover{
    transform: scale(1.05);
  }
`

const CardHeader = styled.header`
  background: #445c95 url(https://www.webankcdn.net/s/hj/www/img/dk03-1d9f1a8ea4.png) no-repeat;
  min-height: 150px;
  padding: 20px;
  font-size: 24px;
  color: #fff;
  font-family: "helvetica neue,hiragino sans gb,microsoft yahei,sans-serif";
`
const CardBody = styled.div`
  padding: 20px;
`

const CardContent = styled.div`
  font-family: "helvetica neue,hiragino sans gb,microsoft yahei,sans-serif";
`
const CTA = styled.button`
  padding: 15px;
  margin-top: 30px;
  background: #3399eb;
  color: #fff;
  min-width: 150px;
  border-radius: 30px;
`

class Card extends PureComponent {
  render() {
    const { onClick, cardDetails } = this.props;
    return (
      <CardContainer>
        <CardHeader>
          Lorem Ipsum
        </CardHeader>
        <CardBody>
          <CardContent>
            { cardDetails || 'Click on CTA' }
          </CardContent>
          <CTA onClick={onClick}>Click Here</CTA>
        </CardBody>
      </CardContainer>
    )
  }
}

export default Card;