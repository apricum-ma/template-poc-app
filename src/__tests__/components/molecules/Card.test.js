import React from 'react';
import { shallow } from 'enzyme';
import Card from '../../../components/molecules/Card';

describe('Card', () => {
  it('should render card', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });
});
