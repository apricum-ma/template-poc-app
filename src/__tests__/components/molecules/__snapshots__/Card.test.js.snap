// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render card 1`] = `ShallowWrapper {}`;
